roles/manage_folder/tasks/main.yml

```diff
@@ -0,0 +1,43 @@
+---
+- name: Check Mandatory Variables Are Defined
+  ansible.builtin.assert:
+    that:
+      - manage_folder_datacenter_name is defined
+      - manage_folder_folder_name is defined and ((manage_folder_folder_name | length) > 0)
+      - manage_folder_hostname is defined
+      - manage_folder_username is defined
+      - manage_folder_password is defined
+    quiet: true
+    fail_msg: Variable must be set when using this role.
+
+- name: Fail If Folder Path Is Absolute
+  ansible.builtin.fail:
+    msg: Folder name should not be absolute. It should be relative to /<datacenter>/<type>
+  when: manage_folder_folder_name[0] == '/'
```

Comment:
Maybe I'm wrong 
I think you mean to fail if manage_folder_folder_name[0] != '/'

Reply:
No, the check is correct. Its making sure the user does not input a path/folder that starts with a slash